content/systems/filecoin_blockchain/struct/block/_index.md

```diff
@@ -48,47 +132,46 @@ A syntactically valid full block must have:
-  - `ElectionPoSt.Candidates` contains well formed `PoStCandidate`s each of which has a `PartialTicket` yielding a winning `ChallengeTicket` in Expected Consensus.
-- a `Timestamp` in seconds that must be
-  - not in the future at time of reception
-  - of the precise value implied implied by the genesis block's timestamp, the network's block time and the block's `Epoch`,
-- all SECP messages correctly signed by their sending actor's worker account key,
-- a `BLSAggregate` signature that signs the array of CIDs of the BLS messages referenced by the block 
-with their sending actor's key.
-- a valid `Signature` over the block header's fields from the block's `Miner` actor's worker account public key.
-
-There is no semantic validation of the messages included in a block beyond validation of their signatures.
-If all messages included in a block are syntactically valid then they may be executed and produce a receipt. 
-
-A chain sync system may perform syntactic and semantic validation in stages in order to minimize unnecessary resource expenditure.
-
+Semantic validation refers to validation that requires reference to information outside the block header and messages themselves, in particular related to the parent tipset and state on which the block is built.
+
+The semantic validation of a block is carried out by the [Chain Synchronization](chainsync) module.
+
+```go
+func (syncer *Syncer) ValidateBlock(ctx context.Context, b *types.FullBlock) error
+```
+
+The semantic validation process for Filecoin blocks includes the following:
+
+- Checks for nils on ElectionProof, Ticket, BlockSig and BLSAggregate
+- Loads parent Tipset via `TipSetKey`. This is needed in order to link to the history of the block being validated.
+- Gets "lookback" tipset and loads the lookback `TipSetState`
+- Gets latest beacon entry
+- Check that a block isn't seen too far into the future (there appears to be an acceptable window into the future for which a block might be accepted)
+
+In parallel, the [ChainSync logic](chainsync) checks that:
+    - the block message is valid
+    - it comes from a valid and not slashed miner
+    - the state root of the parent tipset matches `BlockHeader.ParentStateRoot`
+    - the parent message receipts match `BlockHeader.ParentMessageReceipts`
+
+The ChainSync logic also validates `BlockHeader.ElectionProof`. In order to achieve that it has to:
+        - Get block randomness
+        - Verify election proof
+        - Check that the miner has not been slashed
+        - Check that the miner has power
+        - Check if block was a winner
+ 	  	- Check block signature validation
+        - Signature validation in lib/sigs/sigs.go:
+        ```go
+        func CheckBlockSignature(ctx context.Context, blk *types.BlockHeader, worker address.Address) error```
+	    - Validate drand beacon
+	    - Verify `BlockHeader.WinPoStProof`, which includes:
+    	    - Validate block ticket proofs
+        	- Verify Winning PoSt proof
```

Comment:
This seems like a step backwards to me. It replaces a collection of pure predicates about the block header with Lotus-specific procedural notes, and loses a number of validation rules in the process.

The old list of rules is incomplete, and more should be added. The ElectionPoSt stuff is out of date. The prior state is not great, but the new state is very hand-wavy. What does "valid and not slashed" miner mean exactly? What is "Check if block was a winner"? What's a valid drand beacon?